### PR TITLE
Add server and user_mapping resources

### LINKED ISSUE #198: Add support for foreign server and user mapping
Hello,

We are using the provider very extensively in my organization and we are looking into managing foreign servers and user mappings. 

Foreign Server: https://www.postgresql.org/docs/current/sql-createserver.html
User Mapping: https://www.postgresql.org/docs/current/sql-createusermapping.html

I'm looking into potentially implementing it based on availability. Would this be something that the maintainers would like to support?

Thanks in advance for the hard work on this project!

### PR DESCRIPTION
**Pull request by fabiopaiva**
closes #198

### CODE REVIEW

**Comment by fabiopaiva** [comment]
Hi @cyrilgdn 
Any idea if this can be merged

**Review by PascalBourdier — APPROVED** [review]

**Review by thomasboussekey — APPROVED** [review]

File: postgresql/resource_postgresql_user_mapping.go [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	defer deferredRollback(txn)
+
+	sql := fmt.Sprintf("DROP USER MAPPING FOR %s SERVER %s ", pq.QuoteIdentifier(username), pq.QuoteIdentifier(serverName))
+	if _, err := txn.Exec(sql); err != nil {
+		return err
+	}
+
+	if err = txn.Commit(); err != nil {
+		return fmt.Errorf("Error deleting user mapping: %w", err)
+	}
```
Comment (cyrilgdn) [inline postgresql/resource_postgresql_user_mapping.go:205]: Like above, the transaction can be avoid to simplify the code

**Comment by fabiopaiva** [comment]
@cyrilgdn could you please review it again?

**Comment by fabiopaiva** [comment]
@cyrilgdn, any update?

**Comment by fabiopaiva** [comment]
@cyrilgdn

**Review by cyrilgdn — APPROVED** [review]
Hi,

Sorry again, I was off the project for a while (lack of time).

I'll release your work on next version (this week) :+1:

**Comment by cyrilgdn** [comment]
[v1.18.0](https://registry.terraform.io/providers/cyrilgdn/postgresql/1.18.0) has been released

**Pull request merged**